subsys/audio_module/audio_module.c

```diff
@@ -1117,41 +1137,43 @@ int audio_module_data_tx_rx(struct audio_module_handle *handle_tx,
 		return -EINVAL;
 	}
 
-	if (audio_data_tx->data == NULL || audio_data_tx->data_size == 0 ||
-	    audio_data_rx->data == NULL || audio_data_rx->data_size == 0) {
-		LOG_ERR("One or both of the have invalid output audio data");
-		return -EINVAL;
-	}
-
-	ret = data_tx(NULL, handle_tx, audio_data_tx, NULL);
+	ret = data_tx(NULL, handle_rx, audio_data_tx, NULL);
 	if (ret) {
 		LOG_ERR("Failed to send audio data to module %s, ret %d", handle_tx->name, ret);
 		return ret;
 	}
 
-	LOG_DBG("Wait for message on module %s TX queue", handle_rx->name);
-
-	ret = data_fifo_pointer_last_filled_get(handle_rx->thread.msg_rx, (void **)&msg_rx,
+	ret = data_fifo_pointer_last_filled_get(handle_rx->thread.msg_tx, (void **)&msg_rx,
 						&msg_rx_size, timeout);
 	if (ret) {
-		LOG_ERR("Failed to retrieve audio data from module %s, ret %d", handle_rx->name,
+		LOG_ERR("Failed to retrieve audio data from module %s, ret %d", handle_tx->name,
 			ret);
 		return ret;
 	}
 
-	if (msg_rx->audio_data.data == NULL || msg_rx->audio_data.data_size == 0) {
-		LOG_ERR("Data output buffer too small for received buffer from module %s "
-			"(%d)",
-			handle_rx->name, msg_rx->audio_data.data_size);
-		ret = -EINVAL;
+	if (audio_data_rx->data_size != 0) {
```

Comment:
This new section looks good, but there is duplicated code with the section above. Please factor out.

Reply:
Can we discuss?